.github/workflows/lint.yaml

```diff
@@ -28,6 +28,10 @@ on:
         description: 'concurrency key used to cancel running jobs'
         required: false
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.inputs.concurrency_key || github.sha }}
```

Comment:
If I'm reading right, since all runs from `main` set the concurrency key to `None`, they would all belong in the same concurrency group `${ github.workflow }}-${{ github.sha }}` (because the github sha of the `datadog-agent-macos-build` repo almost never changes), and therefore only one of them can run at the same time.

Are we sure about that? I can see cases (like just before the release freeze, or after the release unfreeze) where the pipelines on `main` are greatly slowed down (or even fail with a timeout on the `datadog-agent` side) if enough `main` pipelines are triggered in a short timeframe.

If not, we could use `datadog_agent_ref` instead of `github.sha`, which (despite its confusing naming) is set by `datadog-agent` to the commit sha of the `datadog-agent` repo (see [lint](https://github.com/DataDog/datadog-agent/blob/aac883cc4f74127aa9e99225898e7b96fc9775f5/.gitlab/source_test/macos.yml#L50), [test](https://github.com/DataDog/datadog-agent/blob/aac883cc4f74127aa9e99225898e7b96fc9775f5/.gitlab/source_test/macos.yml#L19), [build](https://github.com/DataDog/datadog-agent/blob/aac883cc4f74127aa9e99225898e7b96fc9775f5/.gitlab/package_build/dmg.yml#L15) triggers), ensuring that each `main` pipeline gets its own concurrency group [*].

[*] there is one remaining caveat if we do this: multiple pipelines _can_ be run against the same `main` commit (eg. custom builds, pipelines from `integrations-core` | `omnibus-ruby` | `datadog-agent-buildimages` that trigger datadog-agent pipelines to test their own changes). These would still have the same concurrency group, and therefore will be delayed if run concurrently.

Reply:
Oh yes, the commit hash of the Agent repo was my intention. Fixed!

Reply:
We could encode the source of the trigger I suppose but I'm not sure it's worth it at this point. What do you think?

Reply:
> We could encode the source of the trigger I suppose but I'm not sure it's worth it at this point. What do you think?

I think that, for a first pass, it's fine. If this becomes a problem, then what I'd do is make the `datadog-agent` repo send its pipeline id as the concurrency key on `main` and release branches, instead of `None`, therefore ensuring that every `main` / release branch pipeline has its own concurrency group, even if multiple pipelines run on the same commit.